pkg/events/derive/derive.go

```diff
@@ -72,3 +75,7 @@ func makeEventSkeleton(eventID events.ID) eventSkeleton {
 		Params: def.Params,
 	}
 }
+
+func withOriginalContext(event *trace.Event) (trace.Event, error) {
```

Comment:
You should place a comment here explaining the difference between Original and Invoking contexts.